### PR TITLE
Insert with select using streaming call

### PR DESCRIPTION
**Pull request by harshit-gangal**
## Description

This PR does enhancement on the earlier support of insert with select by using the streaming calls for select and there by breaking the insert calls to the tablet into batches. This will also lead to less memory requirement at vttablet and vtgate.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported - Not sure
-   [X] Tests were added or are not required
-   [X] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vtgate/engine/insert.go [hunk 1]
```diff
@@ -243,11 +244,57 @@ func (ins *Insert) TryExecute(vcursor VCursor, bindVars map[string]*querypb.Bind
 
 // TryStreamExecute performs a streaming exec.
 func (ins *Insert) TryStreamExecute(vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool, callback func(*sqltypes.Result) error) error {
-	res, err := ins.TryExecute(vcursor, bindVars, wantfields)
+	if ins.Opcode != InsertSelect {
+		res, err := ins.TryExecute(vcursor, bindVars, wantfields)
```
Comment (systay) [inline go/vt/vtgate/engine/insert.go:248]: Shouldn't primitive to primitive calls always go through the vcursor?

Comment (harshit-gangal) [reply go/vt/vtgate/engine/insert.go:248]: earlier was not needed as the call came through vcursor itself and was delegated, here may be the case.

File: go/test/endtoend/vtgate/queries/insert/main_test.go [hunk 2]
```diff
@@ -97,6 +99,7 @@ func TestMain(m *testing.M) {
 		}
 
 		// Start vtgate
+		clusterInstance.VtGatePlannerVersion = planbuilder.Gen4
```
Comment (frouioui) [inline go/test/endtoend/vtgate/queries/insert/main_test.go:102]: With `gen4` being the new default planner, do we need this line?

File: go/vt/vtgate/planbuilder/insert.go [hunk 3]
```diff
@@ -209,6 +210,10 @@ func buildInsertSelectPlan(ins *sqlparser.Insert, table *vindexes.Table, reserve
 	}
 	eins.Input = plan
 
+	if strings.Contains(plan.GetTableName(), table.Name.String()) {
+		eins.ForceNonStreaming = true
+	}
```
Comment (frouioui) [inline go/vt/vtgate/planbuilder/insert.go:215]: We should add a comment here

Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/insert.go:215]: I added the comment just above this condition.

Comment (harshit-gangal) [reply go/vt/vtgate/planbuilder/insert.go:215]: just realised I did not push that.

File: go/test/endtoend/vtgate/queries/insert/main_test.go (hunk 2 above)
Comment (frouioui) [reply go/test/endtoend/vtgate/queries/insert/main_test.go:102]: Right, makes sense!

**Review by frouioui — APPROVED** [review]
LGTM, left a nit to comment on a piece of code.

**Pull request merged**